integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/util/TckBackendFeatures.java

```diff
@@ -123,4 +123,32 @@ public boolean supportsExtremeScaledNumericValues() {
 		return true;
 	}
 
+	public boolean supportsHighlighterEncoderAtFieldLevel() {
+		return true;
+	}
+
+	public boolean supportsHighlighterUnifiedTypeNoMatchSize() {
+		// Lucene default unified highlighter does not support no-match-size setting.
+		// While in ES a custom highlighter is used that allows for such option.
+		return false;
+	}
+
+	public boolean supportsHighlighterUnifiedTypeMultipleFragmentsAsSeparateItems() {
+		// Lucene default unified highlighter will combine all fragments into one string separating them with `ellipses` parameter.
+		// While in ES a custom highlighter is used that adds each fragment as a separate item to the result list.
+		return false;
+	}
+
+	public boolean supportsHighlighterUnifiedTypeFragmentSize() {
+		// Break iterators from `java.text.BreakIterator` do not allow for such config.
+		// While in ES a custom iterator is available that wraps sentence and word break iterators and is using the max size option.
+		return false;
+	}
+
+	public boolean supportsHighlighterUnifiedTypeMaxAnalyzedOffsetOnFieldsWithTermVector() {
+		// In default unified highlighter if term vectors are set the analyzer is not used when highlighting.
+		// This leads to wrapped analyzer with this setting being not used.
+		// While in ES a custom field highlighter is used that supports this setting.
+		return false;
```

Comment:
Here's a bit more info for a better starting point 😃. You'll probably then want to check https://github.com/apache/lucene/blob/main/lucene/highlighter/src/java/org/apache/lucene/search/uhighlight/UnifiedHighlighter.java#L1228-L1251
this offset strategy is passed into a field highlighter https://github.com/apache/lucene/blob/main/lucene/highlighter/src/java/org/apache/lucene/search/uhighlight/UnifiedHighlighter.java#L1066-L1078
That offset strategy basically tells how much text gets processed:
```
OffsetsEnum offsetsEnums = fieldOffsetStrategy.getOffsetsEnum(reader, docId, content)
 // Highlight the offsetsEnum list against the content to produce Passages.
      Passage[] passages = highlightOffsetsEnums(offsetsEnums);// and breakIterator & scorer
```
If there are no term vectors, then a strategy is using the wrapped analyzer, which limits the "input text". Otherwise, it uses term vectors, and all the text gets processed. So for the user, if they try to limit the amount of text to highlight but simultaneously have term vectors - limit won't work ...